prow/plugins/trigger/generic-comment.go

```diff
@@ -197,20 +198,36 @@ func FilterPresubmits(honorOkToTest bool, gitHubClient GitHubClient, body string
 	return pjutil.FilterPresubmits(filter, changes, branch, presubmits, logger)
 }
 
-func availablePresubmits(githubClient GitHubClient, body, org, repo, branch string, number int, presubmits []config.Presubmit, logger *logrus.Entry) ([]string, error) {
+// availablePresubmits returns 2 sets of presubmits:
+// 1. presubmits that can be run with '/test all' command.
+// 2. presubmits that can be run with their trigger, e.g. '/test job'
+func availablePresubmits(githubClient GitHubClient, body, org, repo, branch string, number int, presubmits []config.Presubmit, logger *logrus.Entry) ([]string, []string, error) {
 	changes := config.NewGitHubDeferredChangedFilesProvider(githubClient, org, repo, number)
-	all := func(p config.Presubmit) (bool, bool, bool) {
-		return true, true, true
+
+	runWithTestAll, err := pjutil.FilterPresubmits(pjutil.TestAllFilter(), changes, branch, presubmits, logger)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var triggerFilters []pjutil.Filter
+	for _, ps := range presubmits {
+		triggerFilters = append(triggerFilters, pjutil.CommandFilter(ps.RerunCommand))
 	}
-	toTest, err := pjutil.FilterPresubmits(all, changes, branch, presubmits, logger)
+	runWithTrigger, err := pjutil.FilterPresubmits(pjutil.AggregateFilter(triggerFilters), changes, branch, presubmits, logger)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
+	}
+
+	var runWithTestAllNames []string
+	for _, ps := range runWithTestAll {
+		runWithTestAllNames = append(runWithTestAllNames, ps.Name)
 	}
-	var available []string
-	for _, pre := range toTest {
-		available = append(available, pre.RerunCommand)
+	var runWithTriggerNames []string
+	for _, ps := range runWithTrigger {
```

Comment:
If I am reading the code correctly this means this list also includes the `runWithTestAll` presubmit. Would it maybe make sense to exclude those?

Reply:
Thank you for the review. Yes, this currently works by listing each individual presubmit that can be run. The exclusion you are referring to is currently done by commeting "Use `/test all` to run all jobs." if the two lists are the same, e.g.:

```
The following commands are available to trigger jobs:

    /test presubmit-1
    /test presubmit-2


Use /test all to run all jobs.
```

Otherwise, jobs will be listed individually under `/test all`, e.g.:

```
The following commands are available to trigger jobs:

    /test presubmit-1
    /test presubmit-2

Use /test all to run the following jobs:

    presubmit-1

```

Please let me know if this needs changing and I will update the PR.